packages/hardhat/contracts/Quest.sol

```diff
@@ -29,18 +32,31 @@ contract Quest {
         IERC20 _rewardToken,
         uint256 _expireTime,
         address _aragonGovernAddress,
-        address payable _fundsRecoveryAddress
+        address payable _fundsRecoveryAddress,
+        IERC20 _depositToken,
+        uint256 _depositAmount,
+        address _questCreator
     ) {
         questTitle = _questTitle;
         questDetailsRef = _questDetailsRef;
         rewardToken = _rewardToken;
         expireTime = _expireTime;
         aragonGovernAddress = _aragonGovernAddress;
         fundsRecoveryAddress = _fundsRecoveryAddress;
+        questCreator = _questCreator;
+        deposit = Models.Deposit(_depositToken, _depositAmount);
+        isDepositReleased = false;
     }
 
     function recoverUnclaimedFunds() external {
         require(block.timestamp > expireTime, "ERROR: Not expired");
+
+        // Restore deposit if not already released
+        if (!isDepositReleased) {
```

Comment:
Problem was that the second time, if rewardToken and depositToken are not the same, the releaseDeposit will fail, so their is a risk of people funding the quest after first reclaim (maybe because didn't refresh his page in the UI), the funds are just stuck in the Quest forever

Reply:
But I can do this check with checking the balance of deposit instead

Reply:
And in the same case, should we allow the creator to reclaim twice ? And have deposit send to him twice ? If I remove the isRelease flag we cant check that he already have reclaim deposit

Reply:
I think it's over complicating things when you should only call this function once anyway. Personally I would just remove this check and leave it at that. It will still work if the tokens are different and it will never need to be called a second time. People don't randomly send tokens to addresses like this contract for no reason.

Reply:
I was thinking at the case when someone fund a quest without refreshing the page and the quest is expired. If there is no possible for him to have his funds back, i dont think we want that.
In my opinion it could potentially result in a bad opinion for Quests product. but maybe I push the thinking too far. Just wanted to be sure we are secure enough as well as user experience compliant.